crates/wallet/src/wallet/mod.rs

```diff
@@ -744,25 +692,53 @@ impl Wallet {
     /// index defined in [BIP32](https://github.com/bitcoin/bips/blob/master/bip-0032.mediawiki),
     /// then returns the last revealed address.
     ///
-    /// # Errors
+    /// **WARNING:** You must persist the staged wallet changes after making this call. For example:
```

Comment:
This comment should say what will happen if we forget to persist changes, but my hunch is that the method will continue to work as expected as long as all staged changes are eventually written before closing the wallet. any thoughts on this?

```suggestion
    /// **WARNING:** To avoid address reuse you must persist the changes resulting from repeated calls to this method before closing the wallet.
```

Reply:
side note that the doc for `reveal_next_address` says

> This will increment the internal derivation index

"internal" is somewhat ambiguous here. Maybe just say it will "increment the keychain's derivation index."

Reply:
I like the wording you suggested since it's more specific about what will happen if you don't persist the changes. I also did some rewording for `reveal_next_address` along the lines you suggested. 

As for moving the warning about persisting updates at the module level, I ike putting it on each affected method so it's super clear which calls need the user to persist changes for, and what will happen if they don't. To that end I added more warnings, hopefully it doesn't look too annoying.